pkg/datarepair/checker/checker.go

```diff
@@ -32,71 +34,63 @@ type Config struct {
 	Interval time.Duration `help:"how frequently checker should audit segments" default:"30s"`
 }
 
-// Checker is the interface for data repair checker
-type Checker interface {
-	// TODO: remove interface
-	Run(ctx context.Context) error
-	IdentifyInjuredSegments(ctx context.Context) (err error)
-	OfflineNodes(ctx context.Context, nodeIDs storj.NodeIDList) (offline []int32, err error)
-	Close() error
-}
-
 // Checker contains the information needed to do checks for missing pieces
-type checker struct {
+type Checker struct {
 	statdb      statdb.DB
 	pointerdb   *pointerdb.Service
 	repairQueue queue.RepairQueue
 	overlay     pb.OverlayServer
 	irrdb       irreparable.DB
 	limit       int
 	logger      *zap.Logger
-	ticker      *time.Ticker
+	Interval    sync2.Cycle
 }
 
 // NewChecker creates a new instance of checker
-func NewChecker(pointerdb *pointerdb.Service, sdb statdb.DB, repairQueue queue.RepairQueue, overlay pb.OverlayServer, irrdb irreparable.DB, limit int, logger *zap.Logger, interval time.Duration) Checker {
+func NewChecker(pointerdb *pointerdb.Service, sdb statdb.DB, repairQueue queue.RepairQueue, overlay pb.OverlayServer, irrdb irreparable.DB, limit int, logger *zap.Logger, interval time.Duration) *Checker {
 	// TODO: reorder arguments
-	return &checker{
+	checker := &Checker{
 		statdb:      sdb,
 		pointerdb:   pointerdb,
 		repairQueue: repairQueue,
 		overlay:     overlay,
 		irrdb:       irrdb,
 		limit:       limit,
 		logger:      logger,
-		ticker:      time.NewTicker(interval),
 	}
+
+	checker.Interval.SetInterval(interval)
+
+	return checker
 }
 
 // Run the checker loop
-func (c *checker) Run(ctx context.Context) (err error) {
+func (checker *Checker) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	for {
-		err = c.IdentifyInjuredSegments(ctx)
+	var group errgroup.Group
```

Comment:
You don't need the group here. You can use `return checker.Interval.Run`. Group is only necessary when it's starting multiple cycles.